lightkurve/factory.py

```diff
@@ -0,0 +1,286 @@
+        elif 'BMJD_OBS' in header and 'FRAMTIME' in header:
+            self.time[frameno] = header['BMJD_OBS'] + header['FRAMTIME']/2./86400.0
+        if 'TIMECORR' in header:
+            self.timecorr[frameno] = header['TIMECORR']
+        if 'CADENCEN' in header:
+            self.cadenceno[frameno] = header['CADENCEN']
+        if 'QUALITY' in header:
+            self.quality[frameno] = header['QUALITY']
+        if 'POS_CORR1' in header:
+            self.pos_corr1[frameno] = header['POS_CORR1']
+        elif 'PTGDIFFX' in header:
+            self.pos_corr1[frameno] = header['PTGDIFFX']
+        if 'POS_CORR2' in header:
+            self.pos_corr2[frameno] = header['POS_CORR2']
+        elif 'PTGDIFFY' in header:
+            self.pos_corr2[frameno] = header['PTGDIFFY']
+
+
+    def _check_data(self):
+        """Check the data before writing to a TPF for any obvious errors."""
+        if len(self.time) != len(np.unique(self.time)):
+            warnings.warn('The factory-created TPF contains cadences with '
+                          'identical TIME values.', LightkurveWarning)
+        if ~np.all(self.time == np.sort(self.time)):
+            warnings.warn('Cadences in the factory-created TPF do not appear '
+                          'to be sorted in chronological order.', LightkurveWarning)
+        if np.sum(self.flux==self.flux) == 0:
+            warnings.warn('The factory-created TPF does not appear to contain '
+                          'non-zero flux values.', LightkurveWarning)
+
+    def get_tpf(self, hdu0_keywords=None, ext_info=None, **kwargs):
+        """Returns a TargetPixelFile object."""
+        if hdu0_keywords is None:
+            hdu0_keywords = {}
+        if ext_info is None:
+            ext_info = {}
+        self._check_data()
+
+        mission = None
+        if 'MISSION' in hdu0_keywords.keys():
+            mission = hdu0_keywords['MISSION']
+
+        from .targetpixelfile import TargetPixelFile, KeplerTargetPixelFile, TessTargetPixelFile
+
+        if (mission=='Kepler') or (mission=='K2'):
+            return KeplerTargetPixelFile(self._hdulist(hdu0_keywords=hdu0_keywords,
+                                        ext_info=ext_info), **kwargs)
+        elif mission=='TESS':
+            return TessTargetPixelFile(self._hdulist(hdu0_keywords=hdu0_keywords,
+                                        ext_info=ext_info), **kwargs)
+        else:
+            return TargetPixelFile(self._hdulist(hdu0_keywords=hdu0_keywords,
+                                        ext_info=ext_info), **kwargs)
+
+    def _hdulist(self, hdu0_keywords, ext_info):
+        """Returns an astropy.io.fits.HDUList object."""
+        return fits.HDUList([self._make_primary_hdu(hdu0_keywords=hdu0_keywords),
+                             self._make_target_extension(ext_info=ext_info),
+                             self._make_aperture_extension()])
+
+    def _header_template(self, extension):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/barentsen/lightkurve/pullRequest?prid=4287604)